### PR TITLE
Brandon createpresenter16

### PR DESCRIPTION
**Pull request by branflake2267**
progress report... I have a majority of the presenter creation done. 

I think I can see the light at the end of the tunnel. I've wrote down the list I need to complete to finish the presenter creation feature.
- fix a couple properties getting sent to template in presenter template creation
- gin module link code generation
- async command to run generation
- warning dialog when things go wrong
- validation of inputs
- code cleanup
- testing template properties generate correctly
- releasing

Not much more left. Seems this stuff is more repeat of the earlier so it should be to difficult now. 

screen shot 2013-10-27 at 9 01 50 am

### CODE REVIEW

**Comment by branflake2267** [comment]
screen shot 2013-10-27 at 9 02 21 am

File: src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java [hunk 1]
```diff
@@ -300,7 +300,10 @@ public PsiClass getNameTokenPsiClass() {
     }
 
     public String getSelectedPackageAndNameAsSubPackage() {
-        return selectedPackage.getQualifiedName() + "." + getName().toLowerCase();
+        if (getName() == null) {
+            setName("");
+        }
+        return selectedPackageRoot.getQualifiedName() + "." + getName().toLowerCase();
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java:316]: nl

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 2]
```diff
@@ -130,26 +124,141 @@ private void run() {
...
+        createPresenterUiHandlers();
+        createPresenterView();
+        createPresenterViewUi();
 
         logger.info("...Creating presenter finished.");
     }
 
+    private void createPresenterViewUi() {
+        RenderedTemplate renderedTemplate = null;
+        if (presenterConfigModel.getNestedPresenter()) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:303]: extract this if/else block to a private method returning the correct RenderedTemplate

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 3]
```diff
@@ -130,26 +124,141 @@ private void run() {
...
+                PsiClass[] createdClasses = createdJavaFile.getClasses();
+                createdPsiClass.setPsiClass(createdClasses[0]);
+                CodeStyleManager.getInstance(project).reformat(createdClasses[0]);
+            }
+        });
+    }
+
+    private void createPresenter() {
+        RenderedTemplate renderedTemplate = null;
+        if (presenterConfigModel.getNestedPresenter()) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:360]: extract

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 4]
```diff
@@ -130,26 +124,141 @@ private void run() {
...
+        PsiClass createdPsiClass = createPsiClass(createdPresenterPackage, renderedTemplate);
+        createdPsiClass.navigate(true);
+    }
+
+    private void createPresenterUiHandlers() {
+        if (!presenterConfigModel.isUseAddUihandlers()) {
+            return;
+        }
+        RenderedTemplate renderedTemplate = null;
+        if (presenterConfigModel.getNestedPresenter()) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:379]: extract

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 5]
```diff
@@ -130,26 +124,141 @@ private void run() {
...
+        } else if (presenterConfigModel.getPopupPresenter()) {
+            renderedTemplate = createdPopupPresenterTemplates.getPresenter();
+        }
+
+        PsiClass createdPsiClass = createPsiClass(createdPresenterPackage, renderedTemplate);
+        createdPsiClass.navigate(true);
+    }
+
+    private void createPresenterUiHandlers() {
+        if (!presenterConfigModel.isUseAddUihandlers()) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:374]:
```
if (presenterConfigModel.isUseAddUihandlers()) {
    do your stuff
}
```

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 6]
```diff
@@ -130,26 +124,141 @@ private void run() {
...
+            renderedTemplate = createdPopupPresenterTemplates.getUihandlers();
+        }
+
+        PsiClass createdPsiClass = createPsiClass(createdPresenterPackage, renderedTemplate);
+        createdPsiClass.navigate(true);
+    }
+
+    private void createPresenterView() {
+        RenderedTemplate renderedTemplate = null;
+        if (presenterConfigModel.getNestedPresenter()) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:394]: extract

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 7]
```diff
@@ -130,26 +124,141 @@ private void run() {
...
+            renderedTemplate = createdPopupPresenterTemplates.getView();
+        }
+
+        PsiClass createdPsiClass = createPsiClass(createdPresenterPackage, renderedTemplate);
+        createdPsiClass.navigate(true);
+    }
+
+    private void createPresenterModule() {
+        RenderedTemplate renderedTemplate = null;
+        if (presenterConfigModel.getNestedPresenter()) {
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:409]: extract

**Pull request merged**